baseapp/msg_service_router.go

```diff
@@ -16,31 +15,52 @@ import (
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 )
 
+type MessageRouterBuilder interface {
+	MessageRouter
+
+	// WithOptios allows to change the default options of the MessageRouter instance after initialization.
+	WithOptions(opts ...MessageRouterOption) MessageRouter
+}
+
 // MessageRouter ADR 031 request type routing
 // https://github.com/cosmos/cosmos-sdk/blob/main/docs/architecture/adr-031-msg-service.md
 type MessageRouter interface {
+	RegisterService(sd *grpc.ServiceDesc, ss interface{})
```

Comment:
We really need the message router to be able able to register services.
Example: https://github.com/cosmos/cosmos-sdk/blob/57bbd6b3773037a20a055a10e173db38a55e5500/runtime/builder.go#L41

Reply:
can we have a interface for consumers in core and the builder interface be in baseapp like it is right now. Modules should not have access to the builder interface. Its another interface but i dont see any other way here

Reply:
but, we might as well not change it now and do it as part of runtime so we can be sure we have the interfaces we want and need in places. Id propose closing this pr but include it in the design phase of runtime

Reply:
Sure! Let's not that. Not too happy with the split interface.